src/main/java/fi/aalto/cs/apluscourses/model/Module.java

```diff
@@ -114,6 +114,10 @@ public URL getUrl() {
     return url;
   }
 
+  public @Nullable ZonedDateTime getTimestamp() {
+    return downloadedAt;
+  }
```

Comment:
There is some synchronization around `downloadedAt` in other methods, so I think it's best to just use the existing `getModuleMetadata` method instead of introducing new method that potentially exposes a race condition

Reply:
...or alternatively synchronize around `versionLock` similarly to `getVersionId`, but I think it's easier to use `getModuleMetadata` and just not worry about it at all